src/Runner/Runner.php

```diff
@@ -159,10 +159,25 @@ public function fix(): array
             return [];
         }
 
-        // @TODO Remove condition for the input argument in 4.0, as it should be required in the constructor
-        return $this->parallelConfig->getMaxProcesses() > 1 && null !== $this->input
-            ? $this->fixParallel()
-            : $this->fixSequential();
+        // @TODO 4.0: Remove condition and it's body, as no longer required when param will be required in the constructor.
+        // This is fallback only for case someone calling `new Runner()` in custom repo and not providing v4 params in v3.
+        if (null === $this->input) {
+            $this->fixSequential();
+        }
+
+        $allowFallbackToSequentialWhenNotEnoughFiles = true;
+
+        if (
+            1 === $this->parallelConfig->getMaxProcesses()
+            || (
+                false === filter_var(getenv('PHP_CS_FIXER_PARALLEL_PREVENT_FALLBACK_TO_SEQUENTIAL_WHEN_NOT_ENOUGH_FILES'), FILTER_VALIDATE_BOOL)
```

Comment:
I don't think we need such a guard. There's no point in running a parallel runner if the amount of files to be analysed is lower than chunk size. We can always fallback to sequential runner and it will be an optimal scenario.

Reply:
when I was working on parallel runner deep review, i frequently run it for 1 or 2 files only.
if it would automatically move me to sequential runner, i would not be able to debug parallel runner.

if not common enough, I'm OK to skip this debug flag (hm... and use `--sequential` flag? 🤦🏻 !)

Reply:
For such debugging you can use chunk size 1 and analyse 2 files (which means 2 worker with 1 file per chunk), so it's still possible to verify parallel with a small amount of files. `--sequential` is for changing the runner completely, regardless of how many files are going to be analysed, so it's slightly different amd have other use cases 🙂.